src/NUnitFramework/framework/Internal/AsyncEnumerableAdapter.NetFramework.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+#if NETFRAMEWORK
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+
+namespace NUnit.Framework.Internal
+{
+    internal static partial class AsyncEnumerableAdapter
+    {
+        private static partial bool TryGetAsyncBlockingEnumerable(object enumerable, out IEnumerable<object?>? result)
+        {
+            result = default;
+
+            var asyncEnumerable = enumerable.GetType().GetInterface("System.Collections.Generic.IAsyncEnumerable`1");
+            if (asyncEnumerable is null)
+                return false;
+
+            var getEnumeratorMethod = asyncEnumerable.GetMethod("GetAsyncEnumerator");
+            if (getEnumeratorMethod is null)
+                return false;
+
+            var enumeratorType = getEnumeratorMethod.ReturnType;
+
+            var currentProperty = enumeratorType.GetProperty("Current");
+            if (currentProperty is null)
+                return false;
+
+            var moveNextMethod = enumeratorType.GetMethod("MoveNextAsync");
+            if (moveNextMethod is null)
+                return false;
+
+            var asyncDisposableType = enumeratorType.GetInterface("System.IAsyncDisposable");
+            if (asyncDisposableType is null)
+                return false;
+
+            var asyncDisposeMethod = asyncDisposableType!.GetMethod("DisposeAsync");
```

Comment:
```suggestion
            var asyncDisposeMethod = asyncDisposableType.GetMethod("DisposeAsync");
```

No need for the null suppression operator.